api/features/views.py

```diff
@@ -187,6 +189,48 @@ def get_serializer_context(self):
 
         return context
 
+    def apply_state_to_queryset(
+        self, query_data: dict[str, typing.Any], queryset: QuerySet[Feature]
+    ) -> QuerySet[Feature]:
+        if not query_data.get("environment"):
+            raise serializers.ValidationError(
+                "Environment is required in order to filter by state search or by state enabled"
+            )
+        state_enabled = query_data["state_enabled"]
+        state_search = query_data["state_search"]
+        environment = query_data["environment"]
+
+        filter_search_q = Q()
+        if state_search is not None:
+            filter_search_q = filter_search_q | Q(
+                feature_states__feature_state_value__string_value__icontains=state_search,
+            )
+
+            if state_search.lower() in {"true", "false"}:
+                boolean_search = state_search.lower() == "true"
+                filter_search_q = filter_search_q | Q(
+                    feature_states__feature_state_value__boolean_value=boolean_search
+                )
+
+            # TODO: Should the integer search be an icontains
+            # equivalent? If so, how?
+            if state_search.isdigit():
+                integer_search = int(state_search)
+                filter_search_q = filter_search_q | Q(
+                    feature_states__feature_state_value__integer_value=integer_search
+                )
+        filter_enabled_q = Q()
+        if state_enabled is not None:
+            filter_enabled_q = filter_enabled_q | Q(
+                feature_states__enabled=state_enabled
+            )
+        queryset = queryset.filter(
+            filter_search_q,
+            filter_enabled_q,
+            feature_states__environment=environment,
+        )
```

Comment:
Hmnm... I don't think this is actually what we want. I think this is going to return incorrect results because of 2 things: 

1. Versioning
2. Overrides

I think that (like the other PR which returns the feature state in the JSON response body), we'll likely need to use `FeatureState.objects.get_live_feature_states` somehow. I'm not entirely sure how we'll do that. We might need to filter that queryset based on the provided parameters and then just filter the main queryset by the feature ids.

Reply:
As we discussed during standup, I've proceeded with what you suggested. Also as discussed, when we bring the PRs together there will be more requests due to some duplication, but only for this method of searching the features page, so the extra traffic will likely to be small.